app/models/agents/website_agent.rb

```diff
@@ -106,24 +106,41 @@ class WebsiteAgent < Agent
 
       Set `http_success_codes` to an array of status codes (e.g., `[404, 422]`) to treat HTTP response codes beyond 200 as successes.
 
+      The `interpolate` option is the way to format extracted data, and add meta information to each resulted payload.  Its value must be a hash, whose key-value pairs are interpolated after extraction and added to each event payload.  e.g.:
+
+          "interpolate": {
```

Comment:
True, those changes would have a performance impact, on mysql we might even hit the string length limit for the event payload if the website is really large. However those limitation already exist if one wants to extract HTML from a POST request, which is only possible by using the `PostAgent -> WebsiteAgent` combination. 

> Perhaps there could be a "propagate and delete immediately" option, which would make an agent instantiate each event only on memory without saving in the database and invoke the receiver agents passing the payload

That could work, an idea that popped into my head is to allow agents to have multiple steps which all are configured in the Agent edit page, but are clearly separated from each other (both in the configuration and the code). The WebsiteAgent could have:

* Request (GET, POST, etc)
* Extract (JSON, HTML or text)
* Interpolate (when feasible)

Depending on the type of the step we could change the description and the configurable options.

That's clearly out of scope for the PR, like I said before, I understand your reasoning behind the `interpolate` option, but would still recommend the `PostAgent -> WebsiteAgent` workaround to users as long as they do not hit a performance limit or a bug caused by too large responses. 
If you think the added complexity is worth not having to configure a second Agent I am ok with the change.